From e09315b80f4fcd9d35b4aa63fe58a5bcd5316372 Mon Sep 17 00:00:00 2001
From: root <polluxsynth@butoba.net>
Date: Fri, 21 Apr 2023 18:37:54 +0100
Subject: [PATCH 1884/1936] Controller.cpp: Set resonance bw when value changed
 (#389)

When resonance bandwidth changed, do not just set the value but
also actually set the underlying relbw member. Failure to do this
will result in the underlying relbw value being uninitialized,
leading to unexpected behavior of the resonance function (usually,
silence) if the resonance bandwidth is set via CC#78 without
first setting the resonance bandwidth depth.
---
 src/Params/Controller.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Params/Controller.cpp b/src/Params/Controller.cpp
index 71da9354..7dc1be0b 100644
--- a/src/Params/Controller.cpp
+++ b/src/Params/Controller.cpp
@@ -379,6 +379,7 @@ void Controller::setresonancecenter(void)
 void Controller::setresonancebw(int value)
 {
     resonancebandwidth.data  = value;
+    setresonancebw();
 }
 
 void Controller::setresonancebw(void)
-- 
2.47.0

